include/libtorrent/packet_pool.hpp

```diff
@@ -153,34 +158,36 @@ namespace libtorrent
 			if (p == nullptr) return;
 
 			packet_ptr pp(p); // takes ownership and may auto free if slab container does not move it
-			std::size_t allocated = pp->allocated;
+			int const allocated = pp->allocated;
 
-			if (allocated <= m_syn_slab.allocate_size) { m_syn_slab.try_push_back(pp); }
-			else if (allocated <= m_mtu_floor_slab.allocate_size) { m_mtu_floor_slab.try_push_back(pp); }
-			else if (allocated <= m_mtu_ceiling_slab.allocate_size) { m_mtu_ceiling_slab.try_push_back(pp); }
+			if (allocated == m_syn_slab.allocate_size) { m_syn_slab.try_push_back(pp); }
+			else if (allocated == m_mtu_floor_slab.allocate_size) { m_mtu_floor_slab.try_push_back(pp); }
```

Comment:
packets may be allocated with not exact mtu_floor size.
like this sequence(548,548,529,529,529,529,529,529,529,529,529,529,529,528,528 (taken from my packet allocation profile log)). less or equal compare gave opportunity to allocate those packets.